squad/ci/backend/lava.py

```diff
@@ -39,20 +40,29 @@ def submit(self, test_job):
                 raise SubmissionIssue(str(fault))
 
     def fetch(self, test_job):
-        data = self.__get_job_details__(test_job.job_id)
+        try:
+            data = self.__get_job_details__(test_job.job_id)
 
-        if data['status'] in self.complete_statuses:
-            yamldata = self.__get_testjob_results_yaml__(test_job.job_id)
-            data['results'] = yaml.load(yamldata)
+            if data['status'] in self.complete_statuses:
+                yamldata = self.__get_testjob_results_yaml__(test_job.job_id)
+                data['results'] = yaml.load(yamldata)
 
-            # fetch logs
-            logs = ""
-            try:
-                logs = self.__get_job_logs__(test_job.job_id)
-            except:
-                self.log_warn("Logs for job %s are not available" % test_job.job_id)
+                # fetch logs
+                logs = ""
+                try:
+                    logs = self.__get_job_logs__(test_job.job_id)
+                except:
+                    self.log_warn("Logs for job %s are not available" % test_job.job_id)
 
-            return self.__parse_results__(data, test_job) + (logs,)
+                return self.__parse_results__(data, test_job) + (logs,)
+        except xmlrpc.client.ProtocolError as error:
```

Comment:
we might get Error or ProtocolError here.

Reply:
the only  exceptions [documented](https://docs.python.org/3/library/xmlrpc.client.html) are `Fault` and `ProtocolError` ... `Error` is the base class for both of them, but in this case we need to differentiate between the two

Reply:
hmm, strange. I'm pretty sure I saw xmlrpc.client.Error in the stack. Anyway, let's have it merged.